src/ucp/wireup/wireup_cm.c

```diff
@@ -734,13 +742,23 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
         return status;
     }
 
-    status = ucp_wireup_connect_local(ep, remote_addr, NULL);
+    status = ucp_wireup_connect_local(ep, remote_addr, NULL, &num_eps_connected);
     if (status != UCS_OK) {
+        ucp_ep_destroy_internal(ep);
         return status;
     }
 
+    if (num_remote_ep_addrs > num_eps_connected) {
+        ucs_error("server received more remote endpoint addresses (%d) than it "
+                  "can connect to (%d)",
+                 num_remote_ep_addrs, num_eps_connected);
+        ucp_ep_destroy_internal(ep);
+        return UCS_ERR_SOME_CONNECTS_FAILED;
```

Comment:
minor: maybe `UCS_ERR_NOT_IMPLEMENTED` or `UCS_ERR_UNSUPPORTED`? to make it clearer that it's temporary limitation?

Reply:
this error prints - 
`[UCX] ucp_ep_create() failed: Failed to connect some of the requested endpoints`
which seems to reflect the actual state so it could give the user (or us) an indication of what's wrong.
but i do see your point. @yosefe ?